File: ts/project/.sdk/tm/rb/test/omni_smoke_test.rb
# Smoke tests for the vendored omni runner itself: a runner that cannot
# FAIL a bad entry would turn every corpus suite vacuously green, so pin
# the failure paths, not just the happy one. (Ruby peer of ts's
# test/omni.test.ts and py's test_omni_smoke.py.)

require "minitest/autorun"
require_relative "../ProjectName_sdk"
require_relative "omni"

class OmniSmokeTest < Minitest::Test
  # A minimal in-memory spec: no fixture file, no OMNI block (lenient v0,
  # like the shared corpus).
  SMOKE_SPEC = {
    "primary" => {
      "smoke" => {
        "basic" => {
          "set" => [
            { "in" => 1, "out" => 2 },
            { "in" => 41, "out" => 42 },
          ],
        },
        "bad" => {
          "set" => [
            { "in" => 1, "out" => 999 },
          ],
        },
        "err" => {
          "set" => [
            { "in" => 0, "err" => "zero refused" },
          ],
        },
      },
    },
  }.freeze

  INC = lambda do |n|
    raise "smoke: zero refused" if 0 == n

    n + 1
  end

  def smoke_pack
    runner = ProjectNameOmni.make_runner(SMOKE_SPEC, ProjectNameSDK.test(nil, nil))
    runner.call("smoke")
  end

  def test_runset_passes_a_correct_subject
    pack = smoke_pack
    pack[:runset].call(pack[:spec]["basic"], INC)
  end

  def test_runset_fails_a_wrong_result_with_omnierror
    pack = smoke_pack
    caught = assert_raises(ProjectNameOmni::OmniError) do
      pack[:runset].call(pack[:spec]["bad"], INC)
    end
    assert_includes caught.message, "result mismatch"
  end

  def test_expected_error_matched_and_missing_error_fails
    pack = smoke_pack

    # The expected error occurs: passes.
    pack[:runset].call(pack[:spec]["err"], INC)

    # The expected error does NOT occur: must fail.
    caught = assert_raises(ProjectNameOmni::OmniError) do
      pack[:runset].call(pack[:spec]["err"], ->(n) { n })
    end
    assert_includes caught.message, "expected error did not occur"
  end
end

File: ts/project/.sdk/tm/php/test/OmniSmokeTest.php
<?php
declare(strict_types=1);

// Smoke tests for the vendored omni runner itself: a runner that cannot
// FAIL a bad entry would turn every corpus suite vacuously green, so pin
// the failure paths, not just the happy one. (PHP peer of ts's
// test/omni.test.ts and py's test_omni_smoke.py.)

require_once __DIR__ . '/../projectname_sdk.php';
require_once __DIR__ . '/Omni.php';

use PHPUnit\Framework\TestCase;
use Voxgig\Omni\OmniError;

class OmniSmokeTest extends TestCase
{
    // A minimal in-memory spec: no fixture file, no OMNI block (lenient
    // v0, like the shared corpus).
    private const SPEC = [
        'primary' => [
            'smoke' => [
                'basic' => [
                    'set' => [
                        ['in' => 1, 'out' => 2],
                        ['in' => 41, 'out' => 42],
                    ],
                ],
                'bad' => [
                    'set' => [
                        ['in' => 1, 'out' => 999],
                    ],
                ],
                'err' => [
                    'set' => [
                        ['in' => 0, 'err' => 'zero refused'],
                    ],
                ],
            ],
        ],
    ];

    private static function inc()
    {
        return function ($n) {
            if (0 === $n) {
                throw new \Exception('smoke: zero refused');
            }
            return $n + 1;
        };
    }

    private static function pack(): array
    {
        $runner = ProjectNameOmni::makeRunner(self::SPEC, ProjectNameSDK::test(null, null));
        return $runner('smoke');
    }

    public function test_runset_passes_a_correct_subject(): void
    {
        $R = self::pack();
        $R['runset']($R['spec']['basic'], self::inc());
        $this->assertTrue(true);
    }

    public function test_runset_fails_a_wrong_result_with_omnierror(): void
    {
        $R = self::pack();
        try {
            $R['runset']($R['spec']['bad'], self::inc());
        } catch (OmniError $err) {
            $this->assertStringContainsString('result mismatch', $err->getMessage());
            return;
        }
        $this->fail('a wrong result did not raise OmniError');
    }

    public function test_expected_error_matched_and_missing_error_fails(): void
    {
        $R = self::pack();

        // The expected error occurs: passes.
        $R['runset']($R['spec']['err'], self::inc());

        // The expected error does NOT occur: must fail.
        try {
            $R['runset']($R['spec']['err'], function ($n) { return $n; });
        } catch (OmniError $err) {
            $this->assertStringContainsString('expected error did not occur', $err->getMessage());
            return;
        }
        $this->fail('a missing expected error did not raise OmniError');
    }
}
